mantidproject/mantid
# Add fitting capability to Eng Diff UI fitting tab

**Description of work.**
- Adds the fit property browser to toolbar.
 - Allows for subtraction of background (so can fit all spectra in a single fit) - for which a new algorithm was made EnggEstimateFocussedBackground.
- Can now load in data as d-spacing or TOF

**To test:**
(1) Perform calibration and focusing as per instructions in PR #28393  
(or download this file [Uploading ENGINX_305761_bank_1.zip…])
(2) Toggle the fit property browser using the 'Fit' button on the toolbar.
(3) Add some functions and fit
(4) Play with background parameters (column 4 onward in table) before and after the subtract background is set to True (it should update the plot if the background if set to True)
(5) Select a row for which subtract background is set to True and click the `Inspect Background' button - it should pop up an external figure which will also update as the parameters are changed.



Partially fixes #27780



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
